pkg/setup/hostname.go

```diff
@@ -32,26 +35,69 @@ func isValidHostname(h string) bool {
 	return hostnameRegex.MatchString(h)
 }
 
-func (e *Environment) setupHostname() error {
-	hostname, err := os.Hostname()
-	if err != nil {
-		return err
-	}
+func checkHostname(hostname string) error {
+	glog.V(4).Infof("Validating hostname %q ...", hostname)
 	if hostname == "" {
+		glog.Errorf("Invalid empty hostname")
 		return fmt.Errorf("empty hostname")
 	}
 	if hostname == "localhost" {
+		glog.Errorf("Invalid hostname: %q", hostname)
 		return fmt.Errorf("invalid hostname: %q", hostname)
 	}
 	if !isValidHostname(hostname) {
+		glog.Errorf("Invalid hostname: %q", hostname)
 		return fmt.Errorf("invalid hostname: %q, %s", hostname, invalidHostnameMessage)
 	}
-	_, err = net.LookupHost(hostname)
+	_, err := net.LookupHost(hostname)
+	if err == nil {
+		glog.V(4).Infof("Valid hostname: %q", hostname)
+		return nil
+	}
+	glog.Errorf("Fail to lookup host: %s", err)
+	return err
+}
+
+func getAWSHostname() (string, error) {
+	glog.V(2).Infof("Trying AWS hostname ...")
+	c := &http.Client{
+		Timeout: time.Second,
+	}
+	resp, err := c.Get("http://169.254.169.254/latest/meta-data/hostname")
```

Comment:
could checking the `/local-hostname` and the `/public-hostname` be of interest ?

Reply:
`/local-hostname` was equal to `/hostname` so I don't think it's useful to add more logic here.
It looks like already being a valid DNS resolvable entry: [source](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html)

**hostname**
The private IPv4 DNS hostname of the instance. In cases where multiple network interfaces are present, this refers to the eth0 device (the device for which the device number is 0).

**local-hostname**
The private IPv4 DNS hostname of the instance. In cases where multiple network interfaces are present, this refers to the eth0 device (the device for which the device number is 0).